src/api/tojs.ts

```diff
@@ -23,37 +21,35 @@ function cache<K, V>(map: Map<any, any>, key: K, value: V, options: ToJSOptions)
 }
 
 function toJSHelper(source, options: ToJSOptions, __alreadySeen: Map<any, any>) {
-    if (!isObservable(source)) return source
+    if (!options.recurseEverything && !isObservable(source)) return source
+
+    if (typeof source !== "object") return source
+
+    // Directly return the Date object itself if contained in the observable
+    if (source instanceof Date) return source
+
+    if (isObservableValue(source)) return toJSHelper(source.get(), options!, __alreadySeen)
+
+    // make sure we track the keys of the object
+    if (isObservable(source)) {
+        keys(source)
+    }
 
     const detectCycles = options.detectCycles === true
 
-    if (
-        detectCycles &&
-        source !== null &&
-        typeof source === "object" &&
-        __alreadySeen.has(source)
-    ) {
+    if (detectCycles && source !== null && __alreadySeen.has(source)) {
         return __alreadySeen.get(source)
     }
 
-    if (isObservableArray(source)) {
+    if (isObservableArray(source) || Array.isArray(source)) {
```

Comment:
why was the second condition dropped here?

Reply:
It is now cloned by the fallback below it seems, but then it is cloned into an object instead of array?

Reply:
Best verify this with a unit test

Reply:
Thanks for raising it. However, if the source is either a plain array or observable array, it will be converted to plain array with the flag set to be true. I've added tests for this scenario.